curvefs/src/client/mds_client.cpp

```diff
@@ -17,37 +17,226 @@
+CURVEFS_ERROR MdsClientImpl::MDSRPCExcutor::DoRPCTask(RPCFunc task) {
+    brpc::Channel channel;
+    std::string mdsaddr = opt_.mdsaddr;
+    int ret = channel.Init(mdsaddr.c_str(), nullptr);
+    if (ret != 0) {
+        LOG(WARNING) << "Init channel failed, addr = " << mdsaddr;
+        return static_cast<CURVEFS_ERROR>(-EHOSTDOWN);
+    }
+
+    brpc::Controller cntl;
+    cntl.set_timeout_ms(opt_.rpcTimeoutMs);
+
+    ret = task(&channel, &cntl);
+    return static_cast<CURVEFS_ERROR>(ret);
+}
+
+
+CURVEFS_ERROR
+MdsClientImpl::Init(const MdsOption &mdsOpt, MDSBaseClient *baseclient) {
+    excutor_.SetOption(mdsOpt);
+    mdsbasecli_ = baseclient;
+    return CURVEFS_ERROR::OK;
+}
+
+#define RPCTask [&](brpc::Channel * channel, brpc::Controller * cntl) -> int // NOLINT
 
 CURVEFS_ERROR MdsClientImpl::CreateFs(const std::string &fsName,
-    const std::string &volumeName) { return CURVEFS_ERROR::OK; }
+                                      uint64_t blockSize,
+                                      const Volume &volume) {
+    auto task = RPCTask {
+        CreateFsResponse response;
+        mdsbasecli_->CreateFs(fsName, blockSize, volume, &response, cntl,
+                              channel);
+        if (cntl->Failed()) {
+            LOG(WARNING) << "CreateFs Failed, errorcode = " << cntl->ErrorCode()
+                         << ", error content:" << cntl->ErrorText()
+                         << ", log id = " << cntl->log_id();
+            return -cntl->ErrorCode();
+        }
+
+        CURVEFS_ERROR retcode = CURVEFS_ERROR::FAILED;
+        FSStatusCode stcode = response.statuscode();
+        FSStatusCode2CurveFSErr(stcode, &retcode);
+        LOG_IF(WARNING, retcode != CURVEFS_ERROR::OK)
+            << "CreateFs: fsname = " << fsName << ", blocksize = " << blockSize
+            << ", volume = " << volume.DebugString()
+            << ", errcode = " << retcode
+            << ", errmsg = " << FSStatusCode_Name(stcode);
+
+        return static_cast<int>(retcode);
+    };
+    return excutor_.DoRPCTask(task);
+}
 
 CURVEFS_ERROR MdsClientImpl::DeleteFs(const std::string &fsName) {
-    return CURVEFS_ERROR::OK; }
+    auto task = RPCTask {
+        DeleteFsResponse response;
+        mdsbasecli_->DeleteFs(fsName, &response, cntl, channel);
+        if (cntl->Failed()) {
+            LOG(WARNING) << "DeleteFs Failed, errorcode = " << cntl->ErrorCode()
+                         << ", error content:" << cntl->ErrorText()
+                         << ", log id = " << cntl->log_id();
+            return -cntl->ErrorCode();
+        }
+
+        CURVEFS_ERROR retcode = CURVEFS_ERROR::FAILED;
+        FSStatusCode stcode = response.statuscode();
+        FSStatusCode2CurveFSErr(stcode, &retcode);
+        LOG_IF(WARNING, retcode != CURVEFS_ERROR::OK)
+            << "DeleteFs: fsname = " << fsName << ", errcode = " << retcode
+            << ", errmsg = " << FSStatusCode_Name(stcode);
+
+        return static_cast<int>(retcode);
```

Comment:
return value may not include in CURVEFS_ERROR enum.

Reply:
it is need. Otherwise, an error will be reported when compiling